src/iter/fold.rs

```diff
@@ -114,4 +118,8 @@ impl<'r, C, ID, F, T> Folder<T> for FoldFolder<'r, C, ID, F>
     fn complete(self) -> C::Result {
         self.base.consume(self.item).complete()
     }
+
+    fn full(&self) -> bool {
+        self.base.full()
+    }
```

Comment:
I guess that depends on whether we view this as merely an optimization to quit early, or if it carries some kind of guarantee of timeliness.  Probably a soft guarantee if any, being parallel and all, since we don't want a ton of synchronization.

Additional consumers/folders that don't have `full` since they're fine with the default: `collect`, `for_each`, `reduce`, `sum`, `product`, and the internal no-op.